awx/api/serializers.py

```diff
@@ -2348,8 +2349,24 @@ class Meta:
         model = Group
 
 
+__all__ = ['CLOUD_PROVIDERS']
+CLOUD_PROVIDERS = []
+
+
+def discover_available_cloud_provider_plugin_names():
+    global CLOUD_PROVIDERS
+    from awx.main.models.inventory import InventorySourceOptions
+
+    plugin_names = list(InventorySourceOptions.injectors.keys())
+
+    CLOUD_PROVIDERS = plugin_names
+
+    return CLOUD_PROVIDERS
```

Comment:
By default, when you do `from awx.api.serializers import *`, that imports all the identifiers considered “public”, meaning everything excluding names starting with a single underscore. When you define an `__all__`, that changes this behavior to only import names listed there. Even if there's no such asterisk-import in the code base, I don't think this list should be redefined to expose `CLOUD_PROVIDERS` instead of everything advertised as “public” before.

Moreover, I'm pretty sure `CLOUD_PROVIDERS` is not imported anywhere and is never used. So it should simply not exist. Assigning it through `global` essentially implements a DIY caching, however it's probably unnecessary. But if the function needs to be called a few times, it's cleaner to decorate it and let the stdlib module care about storing the cached result in an unexposed place.

Here's what I envision being needed here:
```suggestion
from functools import cache


@cache
def discover_available_cloud_provider_plugin_names() -> list[str]:
    """Return a list of cloud plugin names available in runtime.

    The discovery result is cached since it does not change throughout
    the life cycle of the server run.
    
    :returns: List of plugin cloud names.
    :rtype: list[str]
    """
    from awx.main.models.inventory import InventorySourceOptions

    return list(InventorySourceOptions.injectors.keys())
```

Reply:
Seeing the separate module down below, I realized that this is actually supposed to be
```suggestion
from awx.main.utils.plugins import discover_available_cloud_provider_plugin_names
```